qtdialogs/DlgSetupManager.py

```diff
@@ -0,0 +1,1268 @@
+
+         # Column 3: status (Ready/Unlock/Migrate)
+         if isLegacy:
+            def makeMigrateHandler(walletPath):
+               def handler(_):
+                  self.migrateWallet(walletPath)
+               return handler
+            cell, _ = makeButtonCell(
+               self.tr('Migrate'),
+               makeMigrateHandler(fullPath)
+            )
+            self.walletList.setItemWidget(item, 3, cell)
+         elif isEncrypted:
+            def makeUnlockHandler(walletId):
+               def handler(_):
+                  self.unlockWallet(walletId)
+               return handler
+            cell, _ = makeButtonCell(
+               self.tr('Unlock'),
+               makeUnlockHandler(walletId)
+            )
+            self.walletList.setItemWidget(item, 3, cell)
+         else:
+            self.walletList.setItemWidget(item, 3, QtWidgets.QLabel(self.tr('Ready')))
+
+         return True
+      except Exception as e:
+         LOGEXCEPT('Error creating wallet row: %s', str(e))
+         return False
+
+   def _walletStateLabel(self, entry, walletId):
+      """Map bridge state to UI label and flags.
+
+      Returns (label, isLegacy, isEncrypted). Avoids per-row imports by using
+      the bridge-provided enum: unknown(0), legacy(1), migrated(2), encrypted(3),
+      ready(4), loaded(5).
+      """
+      label = self.tr('Ready')
+      isLegacy = False
+      isEncrypted = False
+
+      try:
+         stateVal = entry.state
+         # Handle numeric enums and string fallbacks defensively
+         if isinstance(stateVal, int):
+            if stateVal == 1:
+               label = self.tr('Legacy')
+               isLegacy = True
+            elif stateVal == 3:
+               label = self.tr('Encrypted')
+               isEncrypted = True
+            elif stateVal == 5:
+               # treat loaded as ready for this setup view
+               label = self.tr('Ready')
+            elif stateVal in (2, 4):
+               label = self.tr('Ready')
+            elif stateVal == 0:
+               label = self.tr('Unknown')
+         else:
+            s = str(stateVal).lower()
+            if s == 'legacy':
+               label = self.tr('Legacy')
+               isLegacy = True
+            elif s == 'encrypted':
+               label = self.tr('Encrypted')
+               isEncrypted = True
+            elif s == 'loaded':
+               label = self.tr('Ready')
+            elif s in ('migrated', 'ready'):
+               label = self.tr('Ready')
+            elif s == 'unknown':
+               label = self.tr('Unknown')
+      except Exception:
+         # Default to Ready on any unexpected schema
+         label = self.tr('Ready')
+         isLegacy = False
+         isEncrypted = False
+
+      return (label, isLegacy, isEncrypted)
+
+   def migrateWallet(self, walletPath):
+      """Start migration for a legacy wallet.
+
+      - If a valid main window is available, use the full migration dialog.
```

Comment:
this condition makes no sense. setupmanager is meant to condition the main window, why would it be spawned after the main window, ever?